xarray/coding/variables.py

```diff
@@ -738,3 +779,37 @@ def encode(self, variable: Variable, name: T_Name = None) -> Variable:
 
     def decode(self, variable: Variable, name: T_Name = None) -> Variable:
         raise NotImplementedError()
+
+
+class LiteralTimedelta64Coder(VariableCoder):
+    """Code np.timedelta64 values."""
+
+    def encode(self, variable: Variable, name: T_Name = None) -> Variable:
+        if np.issubdtype(variable.data.dtype, np.timedelta64):
+            dims, data, attrs, encoding = unpack_for_encoding(variable)
+            resolution, _ = np.datetime_data(variable.dtype)
+            attrs["dtype"] = f"timedelta64[{resolution}]"
+            data = duck_array_ops.astype(data, dtype=np.int64, copy=True)
+            return Variable(dims, data, attrs, encoding, fastpath=True)
+        else:
+            return variable
+
+    def decode(self, variable: Variable, name: T_Name = None) -> Variable:
+        if variable.attrs.get("dtype", "").startswith("timedelta64"):
+            dims, data, attrs, encoding = unpack_for_decoding(variable)
+            # overwrite (!) dtype in encoding, and remove from attrs
+            # needed for correct subsequent encoding
+            encoding["dtype"] = attrs.pop("dtype")
```

Comment:
Typically we use the `pop_to()` helper to do this safely, e.g., `dtype = pop_to(encoding, attrs, "dtype", name=name)`

Reply:
Thanks for the heads up!  I somewhat blindly inherited this from the `BooleanCoder`.  I added some better tests around this issue.